test/onigumo_cli_test.exs

```diff
@@ -23,18 +33,41 @@ defmodule OnigumoCLITest do
       assert usage_message_printed?(fn -> Onigumo.CLI.main(["Downloader", "Parser"]) end)
     end
 
-    test("run CLI with invalid switch") do
-      assert usage_message_printed?(fn -> Onigumo.CLI.main(["--invalid"]) end)
+    for switch <- @invalid_switches do
+      test("run CLI with invalid switch #{inspect(switch)}") do
+        assert usage_message_printed?(fn -> Onigumo.CLI.main([unquote(switch)]) end)
+      end
     end
 
     @tag :tmp_dir
     test("run CLI with 'downloader' argument passing cwd", %{tmp_dir: tmp_dir}) do
-      expect(OnigumoDownloaderMock, :main, fn root_path -> root_path end)
+      expect(OnigumoDownloaderMock, :main, fn working_dir -> working_dir end)
 
       File.cd(tmp_dir)
       assert Onigumo.CLI.main(["downloader"]) == tmp_dir
     end
 
+    for switch <- @working_dir_switches do
+      @tag :tmp_dir
+      test("run CLI 'downloader' with #{inspect(switch)} switch", %{tmp_dir: tmp_dir}) do
+        expect(OnigumoDownloaderMock, :main, fn working_dir -> working_dir end)
+
+        assert Onigumo.CLI.main(["downloader", unquote(switch), tmp_dir]) == tmp_dir
+      end
+    end
+
+    test("run CLI 'downloader' with '--working-dir' without any value") do
+      assert usage_message_printed?(fn -> Onigumo.CLI.main(["downloader", "--working-dir"]) end)
+    end
+
+    @tag :tmp_dir
+    test("run CLI 'downloader' with '--working-dir' and value '.'", %{tmp_dir: tmp_dir}) do
+      expect(OnigumoDownloaderMock, :main, fn working_dir -> working_dir end)
+
+      File.cd(tmp_dir)
+      assert Onigumo.CLI.main(["downloader", "--working-dir", "."]) == "."
+    end
```

Comment:
I know it was me who came up with the idea of adding this test. After the discussion at the end of our call, I am no longer sure it tests anything useful. It only verifies that the dot it parsed as a value for the switch, which tests the [OptionParser](https://hexdocs.pm/elixir/OptionParser.html), not anything in Onigumo.

It should be the downloader, or more precisely – the component – or even more precisely – a middleware (It may be a use case for the [use](https://hexdocs.pm/elixir/alias-require-and-import.html#use) macro.) that would ensure proper handling of paths. But maybe [Optimus](https://github.com/funbox/optimus) could handle that, just as Python’s [argparse](https://docs.python.org/3.12/library/argparse.html) has its [FileType](https://docs.python.org/3.12/library/argparse.html#argparse.FileType).

I am sorry about that, but I’d vote for rather not having this test here. We can (and should) revisit path processing later. I won’t however block the PR if the test remains present.

Reply:
Update: [Optimus](https://github.com/funbox/optimus) doesn’t have a file type by itself. It supports custom parsers though, so a reusable one can be written. It is even possible that there already is a [Hex](https://hex.pm/) package for that. Although it [seems](https://hex.pm/packages?search=optimus&sort=recent_downloads) like there isn’t one. Maybe we could publish our work as a side-quest? 😱